pkg/controllers/management/secretmigrator/clusters.go

```diff
@@ -702,60 +703,48 @@ func (m *Migrator) CreateOrUpdatePrivateRegistrySecret(secretName string, rkeCon
 			},
 		}
 	}
+	if existing != nil {
+		err = json.Unmarshal(existing.Data[corev1.DockerConfigJsonKey], &registry)
+		if err != nil {
+			return nil, err
+		}
+	}
 	for _, privateRegistry := range privateRegistries {
 		active[privateRegistry.URL] = struct{}{}
 		if privateRegistry.Password == "" {
 			continue
 		}
-		registry := credentialprovider.DockerConfigJSON{
-			Auths: credentialprovider.DockerConfig{
-				privateRegistry.URL: credentialprovider.DockerConfigEntry{
-					Username: privateRegistry.User,
-					Password: privateRegistry.Password,
-				},
-			},
+		// limitation: if a URL is repeated in the privateRegistries list, it will be overwritten in the registry secret)
```

Comment:
(nit) there's an extra parenthesis that wasn't in the original comment

Reply:
Thanks for catching this. Removed the extra parenthesis and squashed the commits into one.